clients/src/main/java/org/apache/kafka/common/requests/CreateTopicsRequest.java

```diff
@@ -225,20 +237,28 @@ public CreateTopicsRequest(Struct struct, short versionId) {
 
         this.topics = topics;
         this.timeout = struct.getInt(TIMEOUT_KEY_NAME);
+        if (struct.hasField(VALIDATE_ONLY_KEY_NAME))
+            this.validateOnly = struct.getBoolean(VALIDATE_ONLY_KEY_NAME);
+        else
+            this.validateOnly = false;
         this.duplicateTopics = duplicateTopics;
     }
 
     @Override
     public AbstractResponse getErrorResponse(Throwable e) {
-        Map<String, Errors> topicErrors = new HashMap<>();
+        Map<String, CreateTopicsResponse.Error> topicErrors = new HashMap<>();
         for (String topic : topics.keySet()) {
-            topicErrors.put(topic, Errors.forException(e));
+            Errors error = Errors.forException(e);
+            // Avoid populating the error message if it's a generic one
+            String message = error.message().equals(e.getMessage()) ? null : e.getMessage();
```

Comment:
Is it though? Don't they have to map error codes for all requests anyway? The idea was to only include an error message when it provided information over what the error code provides. Do you think sending it back always would be better?